src/pages/RFOX/hooks/useRfoxContext.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  type AccountId,
+  type AssetId,
+  foxOnArbitrumOneAssetId,
+  fromAssetId,
+  thorchainChainId,
+} from '@shapeshiftoss/caip'
+import React, { createContext, useContext, useMemo, useState } from 'react'
+import {
+  selectAccountIdByAccountNumberAndChainId,
+  selectAccountNumberByAccountId,
+  selectFirstAccountIdByChainId,
+} from 'state/slices/portfolioSlice/selectors'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type RFOXContextType = {
+  selectedAssetAccountId: AccountId | undefined
+  selectedAssetId: AssetId
+  stakingAssetAccountId: AccountId | undefined
+  stakingAssetId: AssetId
+  runeMatchingAccountId: AccountId | undefined
+  setSelectedAssetId: (assetId: AssetId) => void
+  setStakingAssetAccountId: React.Dispatch<React.SetStateAction<AccountId | undefined>>
+}
+
+const RFOXContext = createContext<RFOXContextType | undefined>(undefined)
+
+export const RFOXProvider: React.FC<React.PropsWithChildren<{ stakingAssetId: AssetId }>> = ({
+  stakingAssetId = foxOnArbitrumOneAssetId,
+  children,
+}) => {
+  const [selectedAssetId, setSelectedAssetId] = useState<AssetId>(stakingAssetId)
+
+  const stakingAsset = useAppSelector(state => selectAssetById(state, stakingAssetId))
+  // This is required for wallets not supporting BIP44 accounts
+  const defaultStakingAssetAccountId = useAppSelector(state =>
+    selectFirstAccountIdByChainId(state, stakingAsset?.chainId ?? ''),
+  )
+
+  const [stakingAssetAccountId, setStakingAssetAccountId] = useState<AccountId | undefined>(
+    defaultStakingAssetAccountId,
+  )
+
+  const filter = useMemo(
+    () =>
+      stakingAssetAccountId && stakingAssetId
+        ? { assetId: stakingAssetAccountId, accountId: stakingAssetAccountId }
+        : undefined,
+    [stakingAssetAccountId, stakingAssetId],
+  )
+
+  const stakingAssetAccountNumber = useAppSelector(state =>
+    filter ? selectAccountNumberByAccountId(state, filter) : undefined,
+  )
+
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdByAccountNumberAndChainId,
+  )
+
+  const selectedAssetAccountId = useMemo(() => {
+    if (!(filter && stakingAssetAccountNumber !== undefined)) return
+    const accountNumberAccountIds = accountIdsByAccountNumberAndChainId[stakingAssetAccountNumber]
+    const matchingAccountId = accountNumberAccountIds?.[fromAssetId(selectedAssetId).chainId]
+    return matchingAccountId
+  }, [accountIdsByAccountNumberAndChainId, filter, selectedAssetId, stakingAssetAccountNumber])
+
+  const runeMatchingAccountId = useMemo(() => {
```

Comment:
q: do we really need this one? could be default value of `selectedAssetAccountId`, might be better conceptually but I'm wondering so let's ask

Reply:
@NeOMakinG assuming you meant defaulting to the currentl RUNE rewards accountId here - as discussed IRL, indeed may feel a bit odd for users to have matching account number default selection here, removed this logic and instead leveraging currently rewards RUNE AccountId - 95d613a835

MM:

https://jam.dev/c/3757f172-1641-44de-b1ee-f0fbc44a7811

Native:

https://jam.dev/c/80a82c32-45ec-435f-b360-3a0bbf8e2b4a

MM - never staked before account

https://jam.dev/c/ee82a765-11ac-4062-bd86-e41b000dcec5